framework/src/functions/MooseParsedFunctionWrapper.C

```diff
@@ -138,16 +139,32 @@ MooseParsedFunctionWrapper::initialize()
       _scalar_index.push_back(i);
     }
 
+    // Case when a function is bound by the name given in the input values
+    else if (_feproblem.hasFunction(_vals_input[i]))
+    {
+      // The function
+      Function & fn = _feproblem.getFunction(_vals_input[i], _tid);
+
+      // Store a pointer to the function
+      _fns.push_back(&fn);
+
+      // Store a value into _vals
+      _vals.push_back(0);
+
+      // Store the location of this variable
+      _fn_index.push_back(i);
+    }
+
     // Case when a Real is supplied, convert std::string to Real
     else
     {
       // Use istringstream to convert, if it fails produce an error, otherwise add the variable to
       // the _vals variable
       if (!(ss >> tmp) || !ss.eof())
```

Comment:
I know you didn't write this code, but we know now that this isn't sufficient to catch all edge cases. We have a utility that should be called here instead. Perhaps you can fix while you are in here?